lib/charms/hydra/v0/oauth.py

```diff
@@ -201,11 +202,32 @@ def _dump_data(data: Dict, schema: Optional[Dict] = None) -> Dict:
                 ret[k] = json.dumps(v)
             except json.JSONDecodeError as e:
                 raise DataValidationError(f"Failed to encode relation json: {e}")
+        elif isinstance(v, bool):
+            ret[k] = str(v)
         else:
             ret[k] = v
     return ret
 
 
+def strtobool(val: str) -> bool:
```

Comment:
Another **nonblocking** question: I can see this function is very similar to the one in `distutils.util`. Is it possible that we could use that instead?

Reply:
It is taken from `distutils`, `distutils` was marked as deprecated on python `3.10` and `3.11` and it was removed on `3.12` (https://peps.python.org/pep-0632/)